modules/dnn/src/cuda4dnn/csl/cudnn/convolution.hpp

```diff
@@ -225,6 +225,14 @@ namespace cv { namespace dnn { namespace cuda4dnn { namespace csl { namespace cu
                     );
                 }
                 CUDA4DNN_CHECK_CUDNN(cudnnSetConvolutionGroupCount(descriptor, group_count));
+
+#if CUDNN_MAJOR >= 8
+                /* cuDNN 7 and below use FMA math by default. cuDNN 8 includes TF32 Tensor Ops
+                 * in the default setting. TF32 convolutions have lower precision than FP32.
+                 * Hence, we set the math type to CUDNN_FMA_MATH to reproduce old behavior.
+                 */
+                CUDA4DNN_CHECK_CUDNN(cudnnSetConvolutionMathType(descriptor, CUDNN_FMA_MATH));
```

Comment:
> On pre-NVIDIA A100 GPU devices, CUDNN_DEFAULT_MATH and CUDNN_FMA_MATH have the same behavior: Tensor Core kernels will not be selected. With NVIDIA Ampere GPU architecture and CUDA Toolkit 11, CUDNN_DEFAULT_MATH permits TF32 Tensor Core operation and CUDNN_FMA_MATH does not. The TF32 behavior for CUDNN_DEFAULT_MATH can be explicitly disabled by the environment variable NVIDIA_TF32_OVERRIDE=0.
>
> [Source](https://docs.nvidia.com/deeplearning/sdk/cudnn-api/index.html#cudnnMathType_t)

TF32 format is stored as FP32 in memory but uses a 10-bit mantissa during convolution. Hence, it's less precise. This affects only the new CC 8.x GPUs.

Some models like YOLOv4 do not suffer an accuracy drop in FP16. So TF32 might work without issues and give faster performance but it's still not the same as FP32.